spec/scenarios/dunning/campaign_v1_spec.rb

```diff
@@ -0,0 +1,232 @@
+    stub_request(:post, "https://api.stripe.com/v1/payment_intents")
+      .and_return(
+        status: 402,
+        body: lambda { |_req| build_stripe_payment_intent_response.to_json }
+      )
+    stub_request(:post, "https://api.stripe.com/v1/checkout/sessions")
+      .and_return(status: 200, body: {url: "https://stripe.com/checkout/session/cs_test_123"}.to_json)
+
+    WebMock.after_request do |request_signature, response|
+      if request_signature.uri.path.match?(%r{/v1/payment_intents})
+        request_body_hash = if request_signature.url_encoded?
+          Rack::Utils.parse_nested_query(request_signature.body)
+        elsif request_signature.body.json_encoded?
+          JSON.parse(request_signature.body)
+        end
+
+        Jobs::MockStripeWebhookEventJob.perform_later(
+          organization,
+          request_body_hash,
+          JSON.parse(response.body)
+        )
+      end
+    end
+  end
+
+  it do
+    travel_to(DateTime.new(2025, 1, 1, 10)) do
+      create_subscription(
+        {
+          external_customer_id: customer.external_id,
+          external_id: external_subscription_id,
+          plan_code: plan.code
+        }
+      )
+      perform_billing
+
+      expect(webhooks_sent.map { _1["webhook_type"] }).to eq(%w[
+        subscription.started
+        invoice.created
+        invoice.generated
+        invoice.payment_failure
+      ])
+      invoice = customer.invoices.sole
+      expect(invoice.payment_status).to eq("failed")
+      expect(invoice.payment_due_date).to eq(Date.new(2025, 1, 3))
+    end
+
+    # The day after payment_due_date, the invoice should be marked as overdue
+    travel_to(DateTime.new(2025, 1, 4, 13)) do
+      perform_overdue_balance_update
+
+      invoice = customer.invoices.sole
+      expect(invoice).to be_payment_overdue
+      expect(customer.overdue_balance_cents).to eq(149_00)
+
+      # Performing dunning has no effect because the threshold is 150 and we have only 149 overdue
+      perform_dunning
+      expect(customer.payment_requests.count).to eq(0)
+    end
+
+    # Create a one-off invoice to reach the threshold
+    travel_to(DateTime.new(2025, 1, 4, 18)) do
+      create_one_off_invoice(customer, [addon], units: 3)
+      perform_all_enqueued_jobs
+
+      oneoff = customer.invoices.one_off.sole
+      expect(oneoff.payment_status).to eq("failed")
+      expect(oneoff.payment_due_date).to eq(Date.new(2025, 1, 6))
+    end
+
+    travel_to(DateTime.new(2025, 1, 7, 10)) do
+      perform_overdue_balance_update
+      expect(customer.invoices.one_off.sole).to be_payment_overdue
+
+      expect(ActionMailer::Base.deliveries.count).to eq(0)
+      perform_dunning
+      # NOTE: Email is sent twice: first synchronously after Stripe response, and then when the webhook is received
+      expect(ActionMailer::Base.deliveries.count).to eq(2)
+      expect(ActionMailer::Base.deliveries.map(&:subject)).to all eq "Your overdue balance from JC AI"
```

Comment:
Without mocking the webhook, it's almost impossible to notice that.

Reply:
it is expected we send the same email twice or is it a defect?

Reply:
I think it's not expected but we rarely tests with webhook so it's hard to realize. I noticed it as I worked on many stripe feature recently and I wanted to confirm with this test scenario.